rotation.go

```diff
@@ -72,13 +72,15 @@ func (b *backend) populateQueue(ctx context.Context, s logical.Storage) {
 			if !walEntry.LastVaultRotation.IsZero() && walEntry.LastVaultRotation.Before(role.StaticAccount.LastVaultRotation) {
 				// WAL's last vault rotation record is older than the role's data, so
 				// delete and move on
+				log.Debug("deleting outdated WAL", "WAL ID", walEntry.walID)
 				if err := framework.DeleteWAL(ctx, s, walEntry.walID); err != nil {
 					log.Warn("unable to delete WAL", "error", err, "WAL ID", walEntry.walID)
 				}
 			} else {
-				log.Info("adjusting priority for Role")
+				log.Info("found WAL for role",
+					"role", item.Key,
+					"WAL ID", walEntry.walID)
 				item.Value = walEntry.walID
-				item.Priority = time.Now().Unix()
```

Comment:
This change is the main fix

Reply:
Isn't still a good idea to process WALs immediately, particularly in the case where the a manual rotation has happened but failed in the middle? The manual rotation attempt would not be accounted for in the NextRotationTime() calculation.

Reply:
Agreed with @briankassouf.

Reply:
That's a good question. Is there any reason the manual rotation endpoint needs to use the queue at all? It would simplify things a lot if we can just return errors on that endpoint and keep the mechanisms for periodic rotation and manual rotation's retries separate.

I do think that should be the long term direction, but it's maybe more problematic for a patch version.

Reply:
I think the WAL exists to handle errors mid-rotation; which is different than the reason for the queue. Otherwise it's hard to reason about "recovery". I suspect we'll need some kind of WAL approach in either case.

Reply:
The philosophy around using the queue was to standardize how passwords are rotated across all the endpoints. The difference being a manual rotation should be high priority.

Reply:
Thanks for the input. After discussing yesterday, we've taken a fairly different approach to the fix now, instead focussing on making sure we don't allow multiple WALs per role, and therefore trusting that the presence of a WAL implies a manual rotation was requested if it's prior to the scheduled rotation time.